setup.py

```diff
@@ -284,11 +288,27 @@ def get_extensions():
             include_dirs.append(os.path.abspath('./mmcv/ops/csrc/common'))
             include_dirs.append(os.path.abspath('./mmcv/ops/csrc/common/cuda'))
         else:
-            print(f'Compiling {ext_name} without CUDA')
-            op_files = glob.glob('./mmcv/ops/csrc/pytorch/*.cpp') + \
-                glob.glob('./mmcv/ops/csrc/pytorch/cpu/*.cpp')
-            extension = CppExtension
-            include_dirs.append(os.path.abspath('./mmcv/ops/csrc/common'))
+            try:
+                if torch.is_mlu_available():
+                    from torch_mlu.utils.cpp_extension import MLUExtension
+                    define_macros += [('MMCV_WITH_MLU', None)]
+                    mlu_args = os.getenv('MMCV_MLU_ARGS')
+                    extra_compile_args['cncc'] = [mlu_args] if mlu_args else []
+                    op_files = glob.glob('./mmcv/ops/csrc/pytorch/*.cpp')
+                    op_files += glob.glob('./mmcv/ops/csrc/pytorch/cpu/*.cpp')
+                    op_files += glob.glob('./mmcv/ops/csrc/pytorch/mlu/*.cpp')
+                    op_files += glob.glob('./mmcv/ops/csrc/common/mlu/*.mlu')
+                    extension = MLUExtension
+                    include_dirs.append(
+                        os.path.abspath('./mmcv/ops/csrc/common'))
+                else:
+                    print('Cambricon Catch is not available!')
+            except AttributeError:
+                print(f'Compiling {ext_name} without CUDA')
+                op_files = glob.glob('./mmcv/ops/csrc/pytorch/*.cpp') + \
+                    glob.glob('./mmcv/ops/csrc/pytorch/cpu/*.cpp')
+                extension = CppExtension
+                include_dirs.append(os.path.abspath('./mmcv/ops/csrc/common'))
```

Comment:
Maybe we can refactor the block as the following format
```python
#line 290
elif has_method(torch, 'is_mlu_available') and torch.is_mlu_available() or os.getenv('FORCE_MLU', '0') == '1':
    from torch_mlu.utils.cpp_extension import MLUExtension
    define_macros += [('MMCV_WITH_MLU', None)]
    mlu_args = os.getenv('MMCV_MLU_ARGS')
    extra_compile_args['cncc'] = [mlu_args] if mlu_args else []
    op_files = glob.glob('./mmcv/ops/csrc/pytorch/*.cpp')
    op_files += glob.glob('./mmcv/ops/csrc/pytorch/cpu/*.cpp')
    op_files += glob.glob('./mmcv/ops/csrc/pytorch/mlu/*.cpp')
    op_files += glob.glob('./mmcv/ops/csrc/common/mlu/*.mlu')
    extension = MLUExtension
    include_dirs.append(
        os.path.abspath('./mmcv/ops/csrc/common'))
else:
    print(f'Compiling {ext_name} only with CPU')
    op_files = glob.glob('./mmcv/ops/csrc/pytorch/*.cpp') + \
        glob.glob('./mmcv/ops/csrc/pytorch/cpu/*.cpp')
    extension = CppExtension
    include_dirs.append(os.path.abspath('./mmcv/ops/csrc/common'))

Reply:
Done